packages/cardpay-reward-programs/streamlit/views/utils.py

```diff
@@ -40,3 +46,23 @@ def download_csv(payment_list):
         file_name="payment_list.csv",
         mime="text/csv",
     )
+
+
+def view_multiple(program):
+    start_block, end_block = slider_partition(type="two_end")
+    progress = st.progress(0.0)
+    payment_lists = []
+    for i in range(start_block, end_block, program.payment_cycle_length):
+        progress.progress((i - start_block) / (end_block - start_block))
+        tail = min(end_block, i + program.payment_cycle_length)
+        payment_list = program.run(tail, "0x0")
+        if not payment_list.empty:
+            payment_lists.append(payment_list)
+    combined_payment_list = pd.concat(payment_lists)
```

Comment:
The concat happening here still works for everything, just shifted out of the specific rule class.